kolibri/core/analytics/middleware.py

```diff
@@ -31,6 +32,57 @@
     kolibri_process = None
 
 
+def cherrypy_access_log_middleware(get_response):
```

Comment:
I like this feature, but I think it should be optional, mabe based on an option. I don't think we need to add this middleware always, it will mean an extra load of log writting, what could be a performance problem in a crowded server.
What do you think of  using an approach like this https://github.com/learningequality/kolibri/blob/develop/kolibri/core/analytics/middleware.py#L85-L86 ?

Reply:
This logging is important, it should be on by default. When people have crashes or want statistics, this really really helps. We have log rotation to deal with big logs, but adding a log entry for an HTTP request isn't much of an extra load.

Since Kolibri is in a state of active development and troubleshooting, we should not be cheap with the logging. Later on, it can be an easy choice to turn it down.

It's fine that we can switch it off through an extra option. But we already log all static content -- missing the logging of Django views is a big miss for the usefulness of the logs. I think the configurability is separate from this, as it should also affects current behavior.

The current structure of logging and debugging makes it a bit hard, though, and I really want to add a separate PR for 0.14 about this -- especially because I don't think that `--debug` does enough for debugging currently.

Reply:
well, there is already a lot of logging information. If it has worked before without the cherrypy logging it seems that it could be optional. I think we should set it optional at least until we can measure the performance impact of this added logging (`velox` v2 to the rescue ;) )

Reply:
These logging calls have very little impact on performance. What will you be looking for?

You can see the new behavior in the PR description, I took great care to show an example. I can create a log with longer usage of Kolibri, but I think the picture will be the same: Static content log entries eclipse the log entries from calling Django views, which in turn are much more important for the usefulness of logs.

Reply:
There is a longer example and comparison of the new logs further down, also there's now a mechanism in place to filter out specifically noisy HTTP endpoints from the log, in case we need more of that.